python/cudf/cudf/core/column/column.py

```diff
@@ -946,7 +946,9 @@ def as_categorical_column(self, dtype, **kwargs):
         if (
             pd.api.types.pandas_dtype(self.dtype).type in (np.str_, np.object_)
         ) and self.null_count > 0:
+            name = cats._column.name
```

Comment:
These changes are due to the fact that `dropna()` sets the name of the underlying column object to be the same as that of the series.

Reply:
Should this be fixed in the dropna function instead?

Reply:
yeah, probably. Updated